application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -115,6 +150,81 @@ static boolean handleHasBotPermissions(@NotNull String actionVerb,
         return true;
     }
 
+    private static void handleAbsentTarget(@NotNull String actionVerb, @NotNull Interaction event) {
+        event
+            .reply("I can not %s the given user since they are not part of the guild anymore."
+                .formatted(actionVerb))
+            .setEphemeral(true)
+            .queue();
+    }
+
+    /**
+     * Checks whether the given bot and author have enough permission to change the roles of a given
+     * target. For example whether they have enough permissions to add a role to a user.
+     * <p>
+     * If not, it will handle the situation and respond to the user.
+     * <p>
+     * The checks include:
+     * <ul>
+     * <li>the role does not exist on the guild</li>
+     * <li>the target is not member of the guild</li>
+     * <li>the bot or author do not have enough permissions to interact with the target</li>
+     * <li>the bot or author do not have enough permissions to interact with the role</li>
+     * <li>the author does not have the required role for this interaction</li>
+     * <li>the bot does not have the MANAGE_ROLES permission</li>
+     * <li>the given reason is too long</li>
+     * </ul>
+     *
+     * @param role the role to change, or {@code null} if it does not exist on the guild
+     * @param actionVerb the interaction as verb, for example {@code "mute"} or {@code "unmute"}
+     * @param target the target user to change roles from, or {@code null} if the user is not member
+     *        of the guild
+     * @param bot the bot executing this interaction
+     * @param author the author attempting to interact with the target
+     * @param guild the guild this interaction is executed on
+     * @param hasRequiredRole a predicate used to identify required roles by their name
+     * @param reason the reason for this interaction
+     * @param event the event used to respond to the user
+     * @return Whether the bot and the author have enough permission
+     */
+    @SuppressWarnings({"MethodWithTooManyParameters", "BooleanMethodNameMustStartWithQuestion",
+            "squid:S107"})
+    static boolean handleRoleChangeChecks(@Nullable Role role, @NotNull String actionVerb,
+            @Nullable Member target, @NotNull Member bot, @NotNull Member author,
+            @NotNull Guild guild, @NotNull Predicate<? super String> hasRequiredRole,
+            @NotNull CharSequence reason, @NotNull Interaction event) {
+        if (role == null) {
+            event
+                .reply("Can not %s the user, unable to find the corresponding role on this server"
+                    .formatted(actionVerb))
+                .setEphemeral(true)
+                .queue();
```

Comment:
Using an embed here would make it look more neater for the user

Reply:
It is not an embed. Did you fuck up your english and meant to say we should use an embed instead?

Would also have to be aligned with the other moderation commands where we are not using embeds for error messages. And with all other commands as well.

Id say, open a GH issue to discuss this for all commands (tex, tag, ... they all do it like that).

Reply:
Yeah. Even though I am in the top set for English my typing in English is not that good